prometheus-export/src/main/scala/StreamingQuerySource.scala

```diff
@@ -35,8 +35,18 @@ object StreamingQuerySource extends Logging {
 
   /** extract the source name from the description for example
     * "KafkaV2[Subscribe[mpathic-event]]" becomes "KafkaV2"
+    * "org.apache.iceberg.spark.source.SparkMicroBatchStream@32c41681" becomes
+    * "iceberg"
     */
   def extractSourceName(value: String): String = {
+    // if the value starts with org.apache.iceberg.spark.source.SparkMicroBatchStream return iceberg
+    if (
+      value.startsWith("org.apache.iceberg.spark.source.SparkMicroBatchStream")
+    ) {
+      return "IcebergMicroBatchStream"
+    }
+
+    // otherwise split on [ and take the first value
     value.split("\\[").head
```

Comment:
Idiomatic scala tries to make do without `return`. People often argue that it's easier to reason about the flow of the program without early returns, but one very practical argument is that `return` does not interact in a trivial way with lambdas.

```suggestion
    // if the value starts with org.apache.iceberg.spark.source.SparkMicroBatchStream return iceberg
    if (
      value.startsWith("org.apache.iceberg.spark.source.SparkMicroBatchStream")
    ) {
      "IcebergMicroBatchStream"
    } else {
      // otherwise split on [ and take the first value
      value.split("\\[").head
    }
```

Reply:
I am pretty sure the formatter put that in for me